server/service/src/backend_plugin/examples/amc/amc.js

```diff
@@ -0,0 +1,45 @@
+// To upload to server (from this dir):
+// curl --form files='@amc.js' --form "config={};type=application/json" 'http://localhost:8000/plugin?type=AMC&variant-type=BOAJS&code=amc_check'
+
+// TODO type sharing and type sharing
+
+// TODO Should come from settings
+const DAY_LOOKBACK = 800;
+const DAYS_IN_MONTH = 30;
+
+let plugins = {
+  average_monthly_consumption: ({ store_id, filter }) => {
+    const item_ids = filter.item_id.equal_any;
```

Comment:
What if the filter is more complex?
I think our interfaces should be much more closely controlled for plugins?
Passing a filter object seems a bit to easy to screw up? Although I guess it's up to the plugin developer to know what to expect...

It's a bit of a higher level question about how we'll defined out plugin interfaces, would be keen for a discussion...